src/Symfony/Component/String/AbstractString.php

```diff
@@ -605,8 +605,13 @@ public function trimSuffix($suffix): static
         return $str;
     }
 
-    public function truncate(int $length, string $ellipsis = '', bool $cut = true): static
+    public function truncate(int $length, string $ellipsis = '', bool|TruncateCut $cut = TruncateCut::CharStrict): static
     {
+        if (\is_bool($cut)) {
+            trigger_deprecation('symfony/string', '7.2', 'Using the parameter "cut" with a boolean is deprecated and will be removed in 8.0. Use %s enum instead.', TruncateCut::class);
```

Comment:
what about not deprecating? I don't see the need for adding this cost to users

Reply:
I have removed the deprecation :+1: